src/core/model/reducer.ts

```diff
@@ -45,6 +46,21 @@ export const uiSchemaReducer = (
   switch (action.type) {
     case SET_UISCHEMA:
       return buildLinkedUiSchemaTree(action.uiSchema);
+    case ADD_UI_SCHEMA_ELEMENT_TO_LAYOUT:
```

Comment:
we should probably handle `ADD_UI_SCHEMA_ELEMENT_TO_LAYOUT` and `ADD_SCHEMA_ELEMENT_TO_LAYOUT` in the same reducer, I think we should also create a helper function which does the actual ui schema changes to not duplicate this code.

Reply:
so maybe create a new reducer for these actions

Reply:
Hmm I think it's fine to keep it like it is. It's a pure ui schema operation, so it makes sense to add it to the ui schema reducer. Also while the code looks similar I'm not sure we could extract anything worthwhile, except for the errorHandling which takes up a lot of space and will always be the same and occur in every handled action. 

For example like this:
```typescript
const withCloneTree = <R, T1, T2>(
  process: (clonedElement1: T1, clonedElement2: T2) => R,
  [element1, element2, fallback]: [T1, T2, R]
) => {
  const clonedElement1 = cloneTree(element1);
  if (isPathError(clonedElement1)) {
    console.error('An error occured when cloning', element1);
    // Do nothing
    return fallback;
  }
  let clonedElement2: T2 | PathError = element2;
  if (element2) {
    clonedElement2 = cloneTree(element2);
    if (isPathError(clonedElement2)) {
      console.error('An error occured when cloning', element2);
      // Do nothing
      return fallback;
    }
  }
  return process(clonedElement1, clonedElement2);
};
```
which could then be invoked like this:
```typescript
case ADD_UI_SCHEMA_ELEMENT_TO_LAYOUT:
      return withCloneTree(
        (newUiSchema) => {
          const newUIElement = action.uiSchemaElement;
          (newUIElement as LinkedUISchemaElement).parent = newUiSchema;
          (newUiSchema as Layout).elements.splice(
            action.index,
            0,
            newUIElement
          );
          return getRoot(newUiSchema);
        },
        [action.layout as LinkedUISchemaElement, undefined, uiSchema]
      );
```

Reply:
I would group them as they both are operations on the same level, namely the editor

Reply:
Not sure. Wouldn't you then need to group almost all actions into one? For me these are two slightly different actions so I'm fine with keeping them separate.